src/createConfig.ts

```diff
@@ -0,0 +1,20 @@
+import type { SafeKitConfig } from '@safe-global/safe-kit'
+import { Chain as ChainType, custom, http } from 'viem'
+import { CreateConfigParams, SafeConfig } from '@/types/index.js'
+import { isString } from '@/utils.js'
+
+/**
+ * Create a SafeConfig object for the given parameters.
+ * @param param Config parameter object
+ * @returns Config object
+ */
+export function createConfig<
+  Provider extends SafeKitConfig['provider'] = SafeKitConfig['provider'],
+  Chain extends ChainType = ChainType
```

Comment:
I saw that you defined default types in the `@/types/index.js` definition, so why are we defining them again in all the places where we use each type?

Reply:
Added the generics here for getting correct types depending on the inputs. The `SafeConfig.transport` property e.g. has different types depending on the `provider` prop. If the provider is a string, the transport will be a `HttpTransport`, otherwise a `CustomTransport`. 

And regarding the `Chain` generic type here, the idea was that the chain prop in the returned object will be the same as in the passed object. So that way, when I call the function with the `sepolia` object from viem, then the returned object's props will have the specific types if the input object.

So when I call it like this:
```
import { sepolia } from 'viem/chains'

const config = createConfig({ chain: sepolia, ... })
//    ^? type SafeConfig = {
//         chain: {
//           id: 11155111;      <-- would be just a number in the Chain type
//           name: "Sepolia";   <-- would be just a string in the Chain type
//           ...
//         };
//         ....
//       }
```

So the chain object in the resulting object will be more specific.

That was the idea, but if it looks to verbose we can change it of course.